tests/pipelines/run_compiled_model_hotswap.py

```diff
@@ -0,0 +1,151 @@
+    # from test_models_unet_2d_condition.py
+    rank = 4
+    unet_lora_config = LoraConfig(
+        r=rank,
+        lora_alpha=rank,
+        target_modules=["to_q", "to_k", "to_v", "to_out.0"],
+        init_lora_weights=False,
+        use_dora=False,
+    )
+    return unet_lora_config
+
+
+def get_dummy_input():
+    # from UNet2DConditionModelTests
+    batch_size = 4
+    num_channels = 4
+    sizes = (16, 16)
+
+    noise = floats_tensor((batch_size, num_channels) + sizes).to(torch_device)
+    time_step = torch.tensor([10]).to(torch_device)
+    encoder_hidden_states = floats_tensor((batch_size, 4, 8)).to(torch_device)
+
+    return {"sample": noise, "timestep": time_step, "encoder_hidden_states": encoder_hidden_states}
+
+
+def get_lora_state_dicts(modules_to_save):
+    state_dicts = {}
+    for module_name, module in modules_to_save.items():
+        if module is not None:
+            state_dicts[f"{module_name}_lora_layers"] = get_peft_model_state_dict(module)
+    return state_dicts
+
+
+def set_lora_device(model, adapter_names, device):
+    # copied from LoraBaseMixin.set_lora_device
+    for module in model.modules():
+        if isinstance(module, BaseTunerLayer):
+            for adapter_name in adapter_names:
+                module.lora_A[adapter_name].to(device)
+                module.lora_B[adapter_name].to(device)
+                # this is a param, not a module, so device placement is not in-place -> re-assign
+                if hasattr(module, "lora_magnitude_vector") and module.lora_magnitude_vector is not None:
+                    if adapter_name in module.lora_magnitude_vector:
+                        module.lora_magnitude_vector[adapter_name] = module.lora_magnitude_vector[adapter_name].to(
+                            device
+                        )
+
+
+def check_hotswap(do_hotswap):
+    dummy_input = get_dummy_input()
+    unet = get_small_unet()
+    lora_config = get_unet_lora_config()
+    unet.add_adapter(lora_config)
+    torch.manual_seed(42)
+    out_base = unet(**dummy_input)["sample"]
+    # sanity check
+    assert not (out_base == 0).all()
+
+    with tempfile.TemporaryDirectory() as tmp_dirname:
+        lora_state_dicts = get_lora_state_dicts({"unet": unet})
+        StableDiffusionPipeline.save_lora_weights(
+            save_directory=tmp_dirname, safe_serialization=True, **lora_state_dicts
+        )
+        del unet
+
+        unet = get_small_unet()
+        file_name = os.path.join(tmp_dirname, "pytorch_lora_weights.safetensors")
+        unet.load_attn_procs(file_name)
+        unet = torch.compile(unet, mode="reduce-overhead")
+
+        torch.manual_seed(42)
+        out0 = unet(**dummy_input)["sample"]
+
+        # sanity check: still same result
+        atol, rtol = 1e-5, 1e-5
+        assert torch.allclose(out_base, out0, atol=atol, rtol=rtol)
+
+        if do_hotswap:
+            unet.load_attn_procs(file_name, adapter_name="default_0", hotswap=True)
```

Comment:
At this point, the LoRA within `unet` would already be present no? We're not unloading the LoRA above, not sure if it'd affect something.

Reply:
Ah probably that's what we want with hotswapping, nice!